apps/src/templates/rubrics/RubricContainer.jsx

```diff
@@ -134,17 +136,49 @@ export default function RubricContainer({
   };
 
   useEffect(() => {
-    const abort = new AbortController();
     if (!!rubricId && !!sectionId) {
       fetchAiEvaluationStatusAll(rubricId, sectionId).then(response => {
         if (response.ok) {
-          response.json().then(data => setAllAiEvaluationStatus(data));
+          response.json().then(data => {
+            setAiEvalStatusMap(data?.aiEvalStatusMap);
+            delete data.aiEvalStatusMap;
+            setAiEvalStatusCounters(data);
+          });
         }
       });
     }
-    return () => abort.abort();
   }, [rubricId, sectionId]);
 
+  const updateAiEvalStatusForUser = (userId, status) => {
+    setAiEvalStatusMap({
+      ...aiEvalStatusMap,
+      [userId]: status,
+    });
+  };
+
+  const updateAiEvalStatusMap = aiEvalStatusMap => {
+    setAiEvalStatusMap(aiEvalStatusMap);
+  };
+
+  const [hasTeacherFeedbackMap, setHasTeacherFeedbackMap] = useState({});
+
+  const initializeHasTeacherFeedbackMap = allTeacherEvaluationData => {
+    const hasFeedbackMap = {};
+    allTeacherEvaluationData.forEach(userEvalData => {
+      if (userEvalData?.user_id) {
+        hasFeedbackMap[userEvalData.user_id] = userEvalData.eval.length > 0;
+      }
+    });
+    setHasTeacherFeedbackMap(hasFeedbackMap);
+  };
+
+  const updateHasTeacherFeedback = userId => {
```

Comment:
nit: the name of this function makes me assume the value of `hasTeacherFeedback` could be updated to any value, but it actually just sets the value to true for this user. Maybe `onSubmitTeacherFeedback` would better reflect the fact that it's a callback to be executed when feedback is submitted.